be/src/connector/lake_connector.cpp

```diff
@@ -0,0 +1,598 @@
+        if (iter != global_dict_map.end()) {
+            auto& dict_map = iter->second.first;
+            int32_t index = _tablet_schema->field_index(slot->col_name());
+            DCHECK(index >= 0);
+            global_dict->emplace(index, const_cast<GlobalDictMap*>(&dict_map));
+        }
+    }
+    params->global_dictmaps = global_dict;
+    return Status::OK();
+}
+
+Status LakeDataSource::init_unused_output_columns(const std::vector<std::string>& unused_output_columns) {
+    for (const auto& col_name : unused_output_columns) {
+        int32_t index = _tablet_schema->field_index(col_name);
+        if (index < 0) {
+            std::stringstream ss;
+            ss << "invalid field name: " << col_name;
+            LOG(WARNING) << ss.str();
+            return Status::InternalError(ss.str());
+        }
+        _unused_output_column_ids.insert(index);
+    }
+    _params.unused_output_column_ids = &_unused_output_column_ids;
+    return Status::OK();
+}
+
+Status LakeDataSource::init_scanner_columns(std::vector<uint32_t>& scanner_columns) {
+    for (auto slot : *_slots) {
+        DCHECK(slot->is_materialized());
+        int32_t index = _tablet_schema->field_index(slot->col_name());
+        if (index < 0) {
+            std::stringstream ss;
+            ss << "invalid field name: " << slot->col_name();
+            LOG(WARNING) << ss.str();
+            return Status::InternalError(ss.str());
+        }
+        scanner_columns.push_back(index);
+        if (!_unused_output_column_ids.count(index)) {
+            _query_slots.push_back(slot);
+        }
+    }
+    // Put key columns before non-key columns, as the `MergeIterator` and `AggregateIterator`
+    // required.
+    std::sort(scanner_columns.begin(), scanner_columns.end());
+    if (scanner_columns.empty()) {
+        return Status::InternalError("failed to build storage scanner, no materialized slot!");
+    }
+    return Status::OK();
+}
+
+void LakeDataSource::decide_chunk_size() {
+    bool has_huge_length_type = std::any_of(_query_slots.begin(), _query_slots.end(),
+                                            [](auto& slot) { return slot->type().is_huge_type(); });
+    if (_read_limit != -1 && _read_limit < _runtime_state->chunk_size()) {
+        // Improve for select * from table limit x, x is small
+        _params.chunk_size = _read_limit;
+    } else {
+        _params.chunk_size = _runtime_state->chunk_size();
+    }
+    if (has_huge_length_type) {
+        _params.chunk_size = std::min(_params.chunk_size, CHUNK_SIZE_FOR_HUGE_TYPE);
+    }
+}
+
+Status LakeDataSource::init_reader_params(const std::vector<OlapScanRange*>& key_ranges,
+                                          const std::vector<uint32_t>& scanner_columns,
+                                          std::vector<uint32_t>& reader_columns) {
+    const TOlapScanNode& thrift_olap_scan_node = _provider->_t_olap_scan_node;
+    bool skip_aggregation = thrift_olap_scan_node.is_preaggregation;
```

Comment:
Why not share the code with `OlapChunkSize`?

Reply:
I will refactor it later.